src/consumer/stream_consumer.rs

```diff
@@ -49,76 +48,86 @@ unsafe fn disable_nonempty_callback(queue: &NativeQueue) {
     rdsys::rd_kafka_queue_cb_event_enable(queue.ptr(), None, ptr::null_mut())
 }
 
-struct WakerSlab {
-    wakers: Mutex<Slab<Option<Waker>>>,
+/// Data structure to store waker instances from idle streams.
+/// This queue wakes up the wakers in FIFO order.
+/// On registration streams receive an id they can reuse for setting and removing wakers.
+///
+/// Implementation is optimized for adding new wakers and waking all of them.
+/// Removing a waker is less efficient.
+struct WakerQueue {
+    /// Vec containing the set waker instances and the ids of their owning streams
+    waker_queue: Mutex<Vec<(Uuid, Waker)>>,
```

Comment:
Do we really need a `UUID`? We shouldn't expect collisions even if we used 64 bits or less instead of 128, but not sure if there are libraries for this.